DataProcessing/dataHandler.py

```diff
@@ -0,0 +1,125 @@
+import os.path
+import os
+import os.path as osp
+import tempfile
+from argparse import ArgumentParser
+
+import mmcv
+import numpy as np
+from mmtrack.apis import inference_mot, init_model
+from cv2 import imwrite
+import os, glob, shutil
+import tensorflow as tf
+import matplotlib.pyplot as plt
+from sklearn.cluster import KMeans
+from sklearn.metrics import silhouette_score
+import cv2
+
+class DataHandler:
+
+    def __init__(self, 
+                 input_folder_path:str,
+                 output_video_path:str,
+                 config,
+                 checkpoint,
+                 device,
+                 ):
+
+        self.input_path = input_folder_path
+        self.output_path = output_video_path
+        self.config = config
+        self.checkpoint = checkpoint
+        self.device = device
+        self.crop_folder_path = self.output_path + 'crops/'
+        self.cluster_folder_path = self.output_path + 'clusters/'
+        self.cam_id = 2
+
+    def track_persons_and_crop_from_dir(self, capture_index:int=500, acc_threshold:int=0.999) -> None:
+        """
+        capture_index = frame will be captured every capture_index value. input 1 for all frames
+        """
+        def create_crops_folder() -> None:
+            """Creates a crops folder for the given output path"""
+            assert self.output_path
+            if not osp.isdir(self.crop_folder_path):
+                os.mkdir(self.crop_folder_path)
```

Comment:
consider using `os.makedirs(self.crop_folder_path, exist_ok=True)` instead the if statement.
this function creates recursively all missing directories to the given path (if any are missing). exist_ok flag set to `True` means that no exception should be raised in case the folder already exists.
See more information [here](https://docs.python.org/3/library/os.html?highlight=makedirs#os.makedirs)

Reply:
Nice!